uefi-test-runner/src/proto/rng.rs

```diff
@@ -0,0 +1,37 @@
+use uefi::prelude::*;
+use uefi::proto::rng::{Rng, RngAlgorithmType};
+use uefi::table::boot::{BootServices, OpenProtocolAttributes, OpenProtocolParams};
+
+pub fn test(image: Handle, bt: &BootServices) {
+    info!("Running rng protocol test");
+
+    let handle = *bt
+        .find_handles::<Rng>()
+        .expect_success("Failed to get Rng handles")
+        .first()
+        .expect("No Rng handles");
+
+    let rng = bt
+        .open_protocol::<Rng>(
+            OpenProtocolParams {
+                handle,
+                agent: image,
+                controller: None,
+            },
+            OpenProtocolAttributes::Exclusive,
+        )
+        .expect_success("Failed to open Rng protocol");
+    let rng = unsafe { &mut *rng.interface.get() };
+
+    let mut list = [RngAlgorithmType::EMPTY_ALGORITHM; 4];
+
+    rng.get_info(&mut list).unwrap_success();
```

Comment:
One nit here: should get the updated list slice so that only the valid algorithms are printed. E.g.:

```rust
    let mut list = [RngAlgorithmType::EMPTY_ALGORITHM; 4];
    let list = rng.get_info(&mut list).unwrap_success();
    assert!(!list.is_empty());
    info!("Supported rng algorithms : {:?}", list);
```

Reply:
Done